src/NServiceBus.Core/Causation/AttachCausationHeadersBehavior.cs

```diff
@@ -20,35 +25,30 @@ public Task Invoke(IOutgoingLogicalMessageContext context, Func<IOutgoingLogical
         static void SetRelatedToHeader(IOutgoingLogicalMessageContext context, IncomingMessage incomingMessage)
         {
             if (incomingMessage == null)
-            {
                 return;
-            }
 
             context.Headers[Headers.RelatedTo] = incomingMessage.MessageId;
         }
 
-        static void SetConversationIdHeader(IOutgoingLogicalMessageContext context, IncomingMessage incomingMessage)
+        void SetConversationIdHeader(IOutgoingLogicalMessageContext context, IncomingMessage incomingMessage)
         {
             var hasUserDefinedConversationId = context.Headers.TryGetValue(Headers.ConversationId, out var userDefinedConversationId);
 
             if (incomingMessage != null && incomingMessage.Headers.TryGetValue(Headers.ConversationId, out var conversationIdFromCurrentMessageContext))
             {
                 if (hasUserDefinedConversationId)
-                {
                     throw new Exception($"Cannot set the {Headers.ConversationId} header to '{userDefinedConversationId}' as it cannot override the incoming header value ('{conversationIdFromCurrentMessageContext}').");
-                }
 
                 context.Headers[Headers.ConversationId] = conversationIdFromCurrentMessageContext;
                 return;
             }
 
             if (hasUserDefinedConversationId)
-            {
-                // do not override user defined conversation id if no incoming message exists.
                 return;
-            }
 
-            context.Headers[Headers.ConversationId] = CombGuid.Generate().ToString();
+            context.Headers[Headers.ConversationId] = idGenerator(new ConversationIdGeneratorContext());
```

Comment:
When invoking `idGenerator` func, should it be wrapped in `try`/`catch` to provide a meaningful exception instead of just throwing an exception?